### PR TITLE
821 - Bug Fix: No error handling when invalid URL for Video

### LINKED ISSUE #821: No error handling when invalid URL for Video

### PR DESCRIPTION
**Pull request by shirleywbi**
Closes #821 

Adding validation for videos:

- Expected to create node for mp4s and youtube URLs (some public mp4s can be found: https://gist.github.com/jsturgis/3b19447b304616f18657)
- Expected not to work for anything else.

### CODE REVIEW

**Comment by rlnsy** [comment]
Works for YouTube and mp4 cases as described - LGTM! Also, I love the NodeModal tests... gonna be helpful for what I'm working on haha.

File: templates/vue/src/components/__tests__/NodeModal.spec.js
Comment (narendrasss) [inline templates/vue/src/components/__tests__/NodeModal.spec.js:49]: There's some PRs right now, #834 and #836, that once merged would likely change the way you would write these tests. Neither the `WrappedNodeModal` component nor the route mocks would be necessary. Instead you would change this line of code to:

```suggestion
    screen = render(NodeModal, { fixture: oneNodeTapestry }, (_, _, router) => {
        router.push({
          name: names.MODAL,
          params: { nodeId: node.id, type: "add", tab: "content" }
        }) 
    })
```

This makes it closer to the actual API provided by `@testing-library/vue` and makes the test read a bit better. This change would have to wait until those two PRs get merged in however. What do you think @wynnset? Should we keep this as is for now and change it at a later time?

**Review by narendrasss — APPROVED** [review]
Thanks for writing the tests @shirleywbi! The code for the actual bug fix LGTM. I have a small comment on the test but otherwise looks good.

File: templates/vue/src/components/__tests__/NodeModal.spec.js
Comment (wynnset) [reply templates/vue/src/components/__tests__/NodeModal.spec.js:49]: I don't think this is super urgent. I suggest we change the base branch here to `835-improve-test-utils` and wait to merge it in after those PRs are merged in.

Comment (narendrasss) [reply templates/vue/src/components/__tests__/NodeModal.spec.js:49]: Awesome, thanks. @shirleywbi I fixed the merge conflicts, can you make the changes to the tests?

Comment (shirleywbi) [reply templates/vue/src/components/__tests__/NodeModal.spec.js]: @narendrasss Updated with your suggestion. Looks like I might need to do further investigation though because `PermissionsTable` might be missing something. Getting `Cannot read property 'includes' of undefined`.

**Comment by shirleywbi** [comment]
@narendrasss Would you know how to update the Jest configs so that I can import VueYouTubeEmbed? I'm getting this warning without it.
```
[Vue warn]: Unknown custom element: <youtube> - did you register the component correctly? For recursive components, make sure to provide the "name" option.
```

But when I try to import it, it seems that Jest can't parse the package. I'm not particularly sure how to update transform to allow for the package...

```
Jest encountered an unexpected token

    This usually means that you are trying to import a file which Jest cannot parse, e.g. it's not plain JavaScript.

    By default, if Jest sees a Babel config, it will use that to transform your files, ignoring "node_modules".

    Here's what you can do:
     • To have some of your "node_modules" files transformed, you can specify a custom "transformIgnorePatterns" in your config.
     • If you need a custom transformation specify a "transform" option in your config.
     • If you simply want to mock your non-JS modules (e.g. binary assets) you can stub them out with the "moduleNameMapper" config option.

    You'll find more details and examples of these config options in the docs:
    https://jestjs.io/docs/en/configuration.html
```

**Comment by rlnsy** [comment]
@shirleywbi I tried putting in a YouTube URL without the "?" in the id url param
```
https://www.youtube.com/watchv=6NIXwIDRzFo
```
and it was not caught. Also, looks like some of the video cypress tests are failing?

**Comment by shirleywbi** [comment]
> @shirleywbi I tried putting in a YouTube URL without the "?" in the id url param
> 
> ```
> https://www.youtube.com/watchv=6NIXwIDRzFo
> ```
> 
> and it was not caught. Also, looks like some of the video cypress tests are failing?

Weird. It was passing previously. I'll look into it.

**Comment by shirleywbi** [comment]
@rlnsy I can't seem to reproduce your error. Using the link you provided above, I get this: 
image.

**Comment by rlnsy** [comment]
That's weird. For me it produces errors in the console
Screen Shot 2020-11-18 at 1 04 10 PM
.
For reference, "cypress" is just the name of the tapestry. I am using Firefox and I'm on the latest commit (28179684ef7e2781c315dd83cef5ff845af38e73)

**Comment by rlnsy** [comment]
> That's weird. For me it produces errors in the console
> Screen Shot 2020-11-18 at 1 04 10 PM
> .
> For reference, "cypress" is just the name of the tapestry. I am using Firefox and I'm on the latest commit ([2817968](https://github.com/wynnset/tapestry-wp/commit/28179684ef7e2781c315dd83cef5ff845af38e73))

@shirleywbi

**Comment by shirleywbi** [comment]
@rlnsy My first attempt was with Chrome. Just tried it with Firefox and I'm still not coming across this issue.
image

Also, Cypress test for videos are passing locally but seems to have failed above. Any thoughts on why (or how to troubleshoot)?
image

**Comment by shirleywbi** [comment]
@rlnsy Also, mind running e2e on your end to see if the video test is passing?
If it is, then there might be CI issues: https://github.com/cypress-io/cypress-documentation/pull/2571/files
If not, then I'm not sure what's going on.

**Comment by rlnsy** [comment]
@shirleywbi So... I just tested on my other computer (Firefox as well) and it worked with the above URL. I honestly don't know why it didn't work on my other machine but maybe has something to do with what I was working on related to the video URL field in another PR. Anyway seems like a local issue.

Also, tests are passing for me locally as well. We've had issues with cypress in the past where the CI version times out when waiting for certain things to happen, such as closing the modal component. I'm not sure how to increase timeouts though, maybe @narendrasss can chime in.

**Comment by narendrasss** [comment]
@shirleywbi @rlnsy that test has been consistently failing for me actually. I've noticed it in several other branches (most notably #680 and #792). I fixed it while working on 680, which has now been merged in. Let's see if the CI tests pass now.

**Review by wynnset — APPROVED** [review]

**Pull request merged**
